.github/workflows/ci.yml

```diff
@@ -0,0 +1,30 @@
+name: CI process for unscripted
+on:
+  push:
+  pull_request:
+
+permissions:
+  checks: write
+  contents: read
+  id-token: write
+
+jobs:
+  build:
+    runs-on: [self-hosted, small]
+    container:
+      image: drydock-prod.workiva.net/workiva/smithy-runner-dart:1293220
+    steps:
+      - uses: actions/checkout@v3
+
+      - run: |
+          echo "Starting the script sections"
+          dartfmt --set-exit-if-changed -n lib test tool example
+          timeout 5m pub get
+          pub run test
+          tar -czvf unscripted.pub.tgz pubspec.yaml lib README.md LICENSE
+          echo "Script sections completed"
+
+      - uses: Workiva/gha-store-artifacts@refs/tags/v1.0.16
```

Comment:
we can't use this on a public repo

Reply:
This feels like a blocker to the migration of this repo to GHA. Any thoughts @matthewbalvanz-wf ?